modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -50,22 +287,77 @@
 			});
 
 			editor.ui.addRichCombo('TableHeaders', {
-				init() {
-					const headersPrefix = editor.lang.table.headers;
 
-					const headersNone = `${headersPrefix}: ${editor.lang.table.headersNone}`;
-					const headersRow = `${headersPrefix}: ${editor.lang.table.headersRow}`;
-					const headersColumn = `${headersPrefix}: ${editor.lang.table.headersColumn}`;
-					const headersBoth = `${headersPrefix}: ${editor.lang.table.headersBoth}`;
+				// _headersLabels: {
+				// 	[HEADING_BOTH]: `${editor.lang.table.headers}: ${editor.lang.table.headersBoth}`,
+				// 	[HEADING_COL]: `${editor.lang.table.headers}: ${editor.lang.table.headersColumn}`,
+				// 	[HEADING_NONE]: `${editor.lang.table.headers}: ${editor.lang.table.headersNone}`,
+				// 	[HEADING_ROW]: `${editor.lang.table.headers}: ${editor.lang.table.headersRow}`,
+				// },
 
-					this.add(headersNone, headersNone, headersNone);
-					this.add(headersRow, headersRow, headersRow);
-					this.add(headersColumn, headersColumn, headersColumn);
-					this.add(headersBoth, headersBoth, headersBoth);
+				init() {
+
+					// this.add(
+					// 	HEADING_NONE,
+					// 	this._headersLabels[HEADING_NONE],
+					// 	this._headersLabels[HEADING_NONE]
+					// );
+					// this.add(
+					// 	HEADING_ROW,
+					// 	this._headersLabels[HEADING_ROW],
+					// 	this._headersLabels[HEADING_ROW]
+					// );
+					// this.add(
+					// 	HEADING_COL,
+					// 	this._headersLabels[HEADING_COL],
+					// 	this._headersLabels[HEADING_COL]
+					// );
+					// this.add(
+					// 	HEADING_BOTH,
+					// 	this._headersLabels[HEADING_BOTH],
+					// 	this._headersLabels[HEADING_BOTH]
+					// );
+
+					this.add(HEADING_NONE, HEADING_NONE, HEADING_NONE);
+					this.add(HEADING_ROW, HEADING_ROW, HEADING_ROW);
+					this.add(HEADING_COL, HEADING_COL, HEADING_COL);
+					this.add(HEADING_BOTH, HEADING_BOTH, HEADING_BOTH);
+
+					const currentHeading = getHeading(null, editor);
+
+					this.setValue(currentHeading);
 				},
 
 				label: editor.lang.table.headers,
 
+				onClick(selectedHeading) {
+					editor.execCommand(`tableHeading${selectedHeading}`);
+
+					this.setValue(selectedHeading);
+				},
+
+				onRender() {
+					const currentHeading = getHeading(null, editor);
+
+					this.setValue(currentHeading);
```

Comment:
To make sure the value is correct. `this.select` don't have any effect 🤷🏼 🤷🏼 🤷🏼 🤷🏼